runtime_scan/pkg/orchestrator/common/queue.go

```diff
@@ -55,25 +55,25 @@ type Queue[T comparable] struct {
 	// We use a separate map for this instead of reusing inqueue to prevent
 	// calls to Done() removing items from inqueue when they are actually
 	// in the queue slice.
-	processing map[T]struct{}
+	processing map[string]T
 
 	// A map to track items which have been scheduled to be queued at a
 	// later date. We keep track of these items to prevent them being
 	// enqueued earlier than their scheduled time through Enqueue.
-	waitingForEnqueue map[T]struct{}
+	waitingForEnqueue map[string]T
 
 	// A mutex lock which protects the queue from simultaneous reads and
 	// writes ensuring the queue can be used by multiple go routines safely.
 	l sync.Mutex
 }
 
-func NewQueue[T comparable]() *Queue[T] {
+func NewQueue[T ReconcileEvent]() *Queue[T] {
 	return &Queue[T]{
 		itemAdded:         make(chan struct{}),
 		queue:             make([]T, 0),
-		inqueue:           map[T]struct{}{},
-		processing:        map[T]struct{}{},
-		waitingForEnqueue: map[T]struct{}{},
+		inqueue:           make(map[string]T),
+		processing:        make(map[string]T),
+		waitingForEnqueue: make(map[string]T),
```

Comment:
Do we need to store the item (`T`) in `inqueue`, `processing` and `waitingForEnqueue`? Or we can just use `map[string]struct{}{}`?
I didn't found where we are using the value, it seems like we just need a boolean answer.